File: 1608_special_array_with_x_elements_greater_than_or_equal_x/specialArray.py
from typing import List


class Solution:
    def specialArray(self, nums: List[int]) -> int:
        nums.sort(reverse=True)
        n = len(nums)

        i = 0

        while i < n and nums[i] > i:
            i += 1

        if i < n and i == nums[i]:
            return -1
        return i

File: 1608_special_array_with_x_elements_greater_than_or_equal_x/specialArray.go
func specialArray(nums []int) int {
	sort.Slice(nums, func(i, j int) bool {
		return nums[i] > nums[j]
	})

	n := len(nums)
	i := 0

	for i < n && nums[i] > i {
		i += 1
	}

	if i < n && i == nums[i] {
		return -1
	}

	return i
}